app/services/foreman/plugin/role_lock.rb

```diff
@@ -9,17 +9,17 @@ def initialize(plugin_id)
 
       def register_role(name, permissions, role_registry, description = '')
         User.as_anonymous_admin do
-          Role.transaction do
-            role = process_role name, permissions, description
-            role_registry.role_ids << role.id
-          end
+          role = process_role name, permissions, description
+          role_registry.role_ids << role.id
         end
       end
 
       def process_role(name, permissions, description = '')
         role = Role.find_by :name => name
         if role
-          role.update_attribute(:description, description) if role.description != description
+          Role.ignore_locking do
+            role.update_attribute(:description, description) if role.description != description
```

Comment:
Should we just use `update_column` instead so we don't trigger a save? 
also, probably the same fix (either without lock or update column) should be applied on line 29 below

Reply:
If we don't save it here, where would it get saved? Mostly, I find this code confusing so I don't know the answer.

Reply:
`update_column` issues an SQL UPDATE query directly, without triggering active record save with all its callbacks.